src/applications/piezoscan.py

```diff
@@ -98,6 +99,12 @@ def update(self, model):
                                                                    model.xmax + model.step_size,
                                                                    model.current_y + model.step_size,
                                                                    model.ymin])
+        
+        for line in self.ax.lines:
+                if line.get_label() == 'pos':
+                    line.set_marker('')
+        self.ax.plot(self.circle_pos[0],  self.circle_pos[1], 'ro', label = 'pos')
```

Comment:
I think it might be better to not run this loop every time.  The `plot` function returns a line2d object (https://matplotlib.org/stable/api/_as_gen/matplotlib.axes.Axes.plot.html).  Something like this might be better


```
if self.circle_pos is not None:
    if self.position_line2d:
        self.position_line2d.data(self.circle_pos)
    else:
        self.position_line2d = self.ax.plot(self.circle_pos[0], self.circle_pos[1], 'ro', label='pos')
```

Alternatively, if the ScanImage object retains the "position_line2d" handle, then the xy position of that object can be updated without calling the ScanImage.update function.

Reply:
Done